### DISCUSSION TITLE
Is there equivalent of LDM_LIMIT_ONE_SESSION?

### DISCUSSION THREAD

**Discussion by cyberorg** [ltsp/ltsp#252] [Q&A]
We would like to limit users to log into one graphical session from one fat client at a time, is there anything to make that possible?

**Comment by alkisg** [comment]
Hi Jigish,

LDM_LIMIT_ONE_SESSION basically worked for thin clients, not fat clients; as it was able to check for user processes on the server, and also terminate them.

A redesign is needed in order to make it work for fat clients, SSHFS, NFS home, LDAP logins (no SSH connection to the server) etc.
With this design, an "ltsp session" wrapper would be installed in all /usr/share/xsessions/*.desktop files (or a pam hook). The ltsp session would create a ~/.cache/ltsp/session file, and log everything there, like login time, pid, client hostname etc.
The ltsp session would then run mate-session or gnome-session etc, and at the end of the session it would remove ~/.cache/ltsp/session to signify that this session has ended.

That way, if a user would log in twice from a second client, ltsp session could prompt him and ask "continue with this session or abort?".
If the user would select "continue with this session", then ltsp session would temporarily remove the session file, and then the previous session would get notified by some inotify hook and realize that it should exit from there.

Patches welcome; otherwise this is not high in the TODO list and it will need to wait for at least a couple of years...

**Comment by cyberorg** [comment]
Hi Alkis, thanks for the response, I had the similar idea about creating a session file and checking before the launch of the DE, abrupt shutdowns would also have to be handled.

You got anyone willing to implement this if we can offer a small bounty?

OT. I have not worked with the new LTSP yet, will try it some time later this year.
